packages/product-feed-persoo/src/Model/FeedItem/PersooFeedItemFactory.php

```diff
@@ -0,0 +1,127 @@
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Currency\CurrencyFacade $currencyFacade
+     * @param \Shopsys\FrameworkBundle\Model\Product\Collection\ProductUrlsBatchLoader $productUrlsBatchLoader
+     * @param \Shopsys\FrameworkBundle\Model\Category\CategoryRepository $categoryRepository
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductCachedAttributesFacade $productCachedAttributesFacade
+     */
+    public function __construct(
+        protected readonly ProductPriceCalculationForCustomerUser $productPriceCalculationForCustomerUser,
+        protected readonly CurrencyFacade $currencyFacade,
+        protected readonly ProductUrlsBatchLoader $productUrlsBatchLoader,
+        protected readonly CategoryRepository $categoryRepository,
+        protected readonly ProductCachedAttributesFacade $productCachedAttributesFacade,
+    ) {
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Product $product
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Config\DomainConfig $domainConfig
+     * @return \Shopsys\ProductFeed\PersooBundle\Model\FeedItem\PersooFeedItem
+     */
+    public function create(Product $product, DomainConfig $domainConfig): PersooFeedItem
+    {
+        $locale = $domainConfig->getLocale();
+        $rootCategory = $this->categoryRepository->getRootCategory();
+
+        $categories = $product->getCategoriesIndexedByDomainId()[$domainConfig->getId()];
+        $categoryHierarchyNamesByCategoryId = [];
+        $categoryHierarchyIdsByCategoryId = [];
+
+        foreach ($categories as $category) {
+            $categoryHierarchyNames = [];
+            $categoryHierarchyIds = [];
+            $parent = $category->getParent();
+            $categoryHierarchyNames[] = $category->getName($locale);
+            $categoryHierarchyIds[] = $category->getId();
+
+            while ($parent !== null && $parent !== $rootCategory) {
+                $categoryHierarchyIds[] = $parent->getId();
+                $categoryHierarchyNames[] = $parent->getName($locale);
+                $parent = $parent->getParent();
+            }
+
+            $categoryHierarchyNamesByCategoryId[$category->getId()] = implode(' > ', array_reverse($categoryHierarchyNames));
+            $categoryHierarchyIdsByCategoryId[$category->getId()] = implode(':', array_reverse($categoryHierarchyIds));
+        }
+
+        $parameterValuesIndexedByName = [];
+
+        foreach ($this->productCachedAttributesFacade->getProductParameterValues($product) as $productParameterValue) {
+            $parameterValuesIndexedByName[$productParameterValue->getParameter()->getName($locale)] = $productParameterValue->getValue()->getText();
+        }
+
+        return new PersooFeedItem(
+            $product->getId(),
+            $product->getName($domainConfig->getLocale()),
+            $product->getCalculatedSellingDenied(),
```

Comment:
I am a little bit concerned about this, but I am not sure about the specs here. This is something @grossmannmartin is dealing with right now in his task so maybe he already has the answer. What is the ultimate attribute that decides whether the product is in stock or not? When I look into the code, I see the sale exclusion is now a domain-specific attribute so the usage of `$product->getCalculatedSellingDenied()` does not feel correct here :thinking: I am lost TBH

Reply:
This is framework implementation, so it must depend on framework methods. @grossmannmartin is moving this logic to the framework, as we discussed who should move it and Martin will. In that case I am "depending" on his changes and hope they will be done soon, otherwise I would have to make temporary extension of this feed in project, that I do not want to.

Reply:
ok, fair point, thank you for the answer